### PR TITLE
chore(design-tokens-ui): New Design Tokens UI functionality and design

### PR DESCRIPTION
**Pull request by SkaillZ**
### <strong>Pull Request</strong>
Redesign and new functionality according to mock

#### Checklist

- [ ] I have read the CONTRIBUTING doc and I follow the PR guidelines
- [ ] Lint and unit tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)

### CODE REVIEW

File: package-lock.json [hunk 1]
```diff
@@ -33782,28 +33782,28 @@
           "dependencies": {
             "abbrev": {
               "version": "1.1.1",
-              "resolved": false,
```
Comment (ffriedl89) [inline package-lock.json:33785]: Please revert these changes in the package.lock, this will otherwise be included again in another PR

File: apps/design-tokens-ui/src/app/components/nav/nav.component.scss [hunk 2]
```diff
@@ -25,6 +25,21 @@ $logo-height: 28px;
   width: $default-width;
   transition: width 200ms, background-color 200ms, color 200ms;
 
+  .design-tokens-ui-nav-logo {
+    width: 28px;
```
[the remark this answers is not in the document]
Comment (tomheller) [reply apps/design-tokens-ui/src/app/components/nav/nav.component.scss:29]: @lukasholzer we don't need to write one, there already is one out there:
https://github.com/AndyOGo/stylelint-declaration-strict-value

File: apps/design-tokens-ui/src/app/components/nav/nav.component.scss [hunk 3]
```diff
@@ -38,8 +53,12 @@ $logo-height: 28px;
       fill: var(--color-neutral-50);
     }
 
-    .design-tokens-ui-nav-logo img {
-      filter: brightness(0);
+    .design-tokens-ui-nav-logo {
+      width: 100%;
+
+      img {
+        filter: brightness(0);
```
Comment (tomheller) [inline apps/design-tokens-ui/src/app/components/nav/nav.component.scss:60]: I'm pretty sure there is a version of the logo out there, that we do not need to recolor with css-filters, but it's fine for now.

File: apps/design-tokens-ui/src/pages/theme-detail/components/distribution-curve/distribution-curve.component.ts [hunk 4]
```diff
@@ -108,17 +93,10 @@ export class ConstrastDistributionCurveComponent
...
-      .pipe(takeUntil(this._destroy$))
-      .subscribe(() => {
-        this.distributionsChange.emit(this._calculateDistributions());
-      });
   }
 
-  ngOnChanges(): void {
-    this._stateChanges$.next();
+  ngOnChanges(changes: SimpleChanges): void {
+    this._stateChanges$.next(changes.generationOptions.currentValue);
```
Comment (tomheller) [inline apps/design-tokens-ui/src/pages/theme-detail/components/distribution-curve/distribution-curve.component.ts:99]: You could check if generationOptions actually changed here, and only next it if it is different.

File: apps/design-tokens-ui/src/pages/theme-detail/components/easing-type-select/easing-type-select.component.scss [hunk 5]
```diff
@@ -0,0 +1,10 @@
+@import '~@dynatrace/fluid-design-tokens/global/spacing';
+
+:host {
+  display: block;
+  margin: $fluid-spacing--x-small 0 $fluid-spacing--medium;
```
Comment (tomheller) [inline apps/design-tokens-ui/src/pages/theme-detail/components/easing-type-select/easing-type-select.component.scss:5]: there is also a fluid-spacing--0 😉

**Comment by SkaillZ** [comment]
No idea why lint is still failing, should have been fixed since I've rebased master

File: apps/design-tokens-ui/src/services/app-theme/app-theme.service.ts [hunk 6]
```diff
@@ -0,0 +1,112 @@
...
+
+  private _themeChangeSubject$ = new Subject<string>();
+
+  constructor(private _router: Router) {}
+
+  /** Initializes event listers for app theming */
+  initialize(): void {
+    this._router.events
+      .pipe(filter((event) => event instanceof ActivationStart))
+      .subscribe((event: ActivationStart) => {
```
Comment (lukasholzer) [inline apps/design-tokens-ui/src/services/app-theme/app-theme.service.ts:55]: please unsubscribe

File: apps/design-tokens-ui/src/pages/themes/themes.component.ts [hunk 7]
```diff
@@ -0,0 +1,39 @@
...
+  selector: 'design-tokens-ui-themes',
+  templateUrl: './themes.component.html',
+  styleUrls: ['./themes.component.scss'],
+})
+export class ThemesComponent {
+  _themes$: Observable<Theme[]>;
+
+  constructor(private _paletteSourceService: PaletteSourceService) {
+    this._themes$ = _paletteSourceService.themes$;
+  }
```
Comment (lukasholzer) [inline apps/design-tokens-ui/src/pages/themes/themes.component.ts:33]:
```suggestion
  _themes$: Observable<Theme[]> = this._paletteSourceService.themes$;;

  constructor(private _paletteSourceService: PaletteSourceService) { }
```

File: apps/design-tokens-ui/src/services/app-theme/app-theme.service.ts [hunk 8]
```diff
@@ -0,0 +1,112 @@
...
+  private _updateTheme(): void {
+    const theme = this._resolveTheme();
+    if (theme === this._currentTheme) {
+      return;
+    }
+
+    this._currentTheme = theme;
+
+    const currentThemeClass = [
+      ...(document.body.classList as any),
```
Comment (lukasholzer) [inline apps/design-tokens-ui/src/services/app-theme/app-theme.service.ts:85]: please inject document via DI (`@Inject(DOCUMENT)...`)

File: apps/design-tokens-ui/src/services/palette/palette-source.service.ts [hunk 9]
```diff
@@ -15,95 +15,471 @@
...
+          })),
+      })),
+    };
+
+    this.modifyState(() => initialState);
+
+    // Only regenerate all palette colors if relevant properties changed
+    this.theme$
+      .pipe(
+        share(),
```
Comment (lukasholzer) [inline apps/design-tokens-ui/src/services/palette/palette-source.service.ts:141]: puth the share operator in the theme pipe on line 92

File: apps/design-tokens-ui/src/services/palette/palette-source.service.ts [hunk 10]
```diff
@@ -15,95 +15,471 @@
...
+        distinctUntilChanged(this._canSkipPaletteGeneration),
+        filter<Theme>(Boolean),
+        debounceTime(200),
+      )
+      .subscribe((theme) => {
```
Comment (lukasholzer) [inline apps/design-tokens-ui/src/services/palette/palette-source.service.ts:146]: please unsubscribe

File: apps/design-tokens-ui/src/services/palette/palette-source.service.ts [hunk 11]
```diff
@@ -15,95 +15,471 @@
...
+        for (const palette of theme.palettes) {
+          this.regeneratePaletteColors(palette);
+        }
+        this.overrideStylesForThemePalettes();
+      });
+
+    // Regenerate palette on changes
+    this.palette$
+      .pipe(share(), filter<Palette>(Boolean), debounceTime(200))
+      .subscribe((palette) => {
```
Comment (lukasholzer) [inline apps/design-tokens-ui/src/services/palette/palette-source.service.ts:156]: please unsubscribe

File: apps/design-tokens-ui/src/services/palette/palette-source.service.ts [hunk 12]
```diff
@@ -15,95 +15,471 @@
...
+        this.regeneratePaletteColors(palette);
+        this.overrideStylesForThemePalettes();
+      });
+
+    router.events
+      .pipe(
+        filter((evt) => evt instanceof ActivationStart),
+        map((evt: ActivationStart) => evt.snapshot.paramMap),
+      )
+      .subscribe((params: ParamMap) => {
```
Comment (lukasholzer) [inline apps/design-tokens-ui/src/services/palette/palette-source.service.ts:166]: please unsubscribe

File: libs/design-tokens-ui/shared/tsconfig.lib.prod.json [hunk 13]
```diff
@@ -0,0 +1,6 @@
+{
+  "extends": "./tsconfig.lib.json",
+  "angularCompilerOptions": {
+    "enableIvy": false
```
Comment (lukasholzer) [inline libs/design-tokens-ui/shared/tsconfig.lib.prod.json:4]: NIT: please remove

File: libs/design-tokens-ui/shared/README.md [hunk 14]
```diff
@@ -0,0 +1,7 @@
+# design-tokens-ui-shared
+
+This library was generated with [Nx](https://nx.dev).
+
+## Running unit tests
+
+Run `nx test design-tokens-ui-shared` to execute the unit tests.
```
Comment (lukasholzer) [inline libs/design-tokens-ui/shared/README.md:7]: NIT: Maybe a little bit more descriptive and specific shot explanation about the use case

File: apps/design-tokens-ui/src/services/style-overrides/style-overrides.service.ts [hunk 15]
```diff
@@ -0,0 +1,66 @@
...
+ * Utility for live overwriting styles in the UI
+ */
+@Injectable({
+  providedIn: 'root',
+})
+export class StyleOverridesService {
+  private _overrideStyleElement: HTMLStyleElement;
+
+  constructor() {
+    let styleElement = document.createElement('style');
```
Comment (lukasholzer) [inline apps/design-tokens-ui/src/services/style-overrides/style-overrides.service.ts:29]: please inject Doucment via DI

File: apps/design-tokens-ui/src/services/palette/palette-source.service.ts [hunk 16]
```diff
@@ -15,95 +15,471 @@
...
+        ? palette.tokenData.generationOptions
+        : themeGenerationOptions.get(theme.name),
+    }));
+
+    const yaml = stringify({
+      meta,
+      aliases,
+    });
 
     const element = document.createElement('a');
```
Comment (lukasholzer) [inline apps/design-tokens-ui/src/services/palette/palette-source.service.ts:479]: please inject via DI

File: apps/design-tokens-ui/src/services/palette/palette-source.service.ts [hunk 17]
```diff
@@ -115,16 +491,35 @@ export class PaletteSourceService {
     document.body.removeChild(element);
```
Comment (lukasholzer) [inline apps/design-tokens-ui/src/services/palette/palette-source.service.ts:491]: please inject via DI

**Comment by lukasholzer** [comment]
@tomheller build was failing again on github actions with `permissions denied ng` – I retriggered it (have you encountered this problem before?)

**Comment by tomheller** [comment]
@lukasholzer Yes, this happened from time to time. I have no idea why this fails.
